scripts/validate_docstrings.py

```diff
@@ -389,6 +389,18 @@ def deprecated(self):
     def mentioned_private_classes(self):
         return [klass for klass in PRIVATE_CLASSES if klass in self.raw_doc]
 
+    @property
+    def section_order(self):
+        sections = ["Summary", "Parameters", "Returns",
+                    "Yields", "See Also", "Notes", "Examples"]
+        section_index = []
+        for section in sections:
+            if section not in self.raw_doc:
+                continue
+            else:
+                section_index.append(self.raw_doc.index(section))
+        return section_index
```

Comment:
this implementation doesn't seem realiable, if the summary contains for example `Returns...` this will return the wrong indices, right?

Also, the name of the propertyis not very intuitive on what it does, and if something like this needs to be implemented, we should have tests for it.

I think we use numpydoc for parsing the docstring. May be there is something there that can be used?

Reply:
@datapythonista yup that’s true - I wanted to use numpydoc for parsing but wasn’t sure it would work for the summary and short summary sections as there is usually no title/header to parse the content.... trying to find a workaround